webclient/src/components/KnownHosts/KnownHosts.tsx

```diff
@@ -1,85 +1,208 @@
+      await hostDTO.save();
+
+      setHostsState(s => ({
+        ...s,
+        hosts: s.hosts.map(h => h.id === id ? hostDTO : h),
+      }));
+    } else {
+      const newHost: Host = { name, host, port, editable: true };
+      newHost.id = await HostDTO.add(newHost) as number;
+
+      setHostsState(s => ({
+        ...s,
+        hosts: [...s.hosts, newHost],
+        selectedHost: newHost,
+      }));
+    }
+
+    closeKnownHostDialog();
+  };
+
+  const updateLastSelectedHost = (hostId): Promise<any[]> => {
+    return HostDTO.getAll().then(hosts =>
+      hosts.map(async host => {
+        if (host.id === hostId) {
+          host.lastSelected = true;
+          return await host.save();
+        }
+
+        if (host.lastSelected) {
+          host.lastSelected = false;
+          return await host.save();
         }
-      </Select>
-    </FormControl>
+
+        return host;
+      })
+    );
+  };
+
+  return (
+    <div>
+      <FormControl variant='outlined' className={'KnownHosts ' + classes.root}>
+        { touched && (
+          <div className="KnownHosts-validation">
+            {
+              (error &&
+                <div className="KnownHosts-error">
+                  {error}
+                  <ErrorOutlinedIcon style={{ fontSize: 'small', fontWeight: 'bold' }} />
+                </div>
+              ) ||
+
+              (warning && <div className="KnownHosts-warning">{warning}</div>)
+            }
+          </div>
+        ) }
+
+        <InputLabel id='KnownHosts-select'>Host</InputLabel>
+        <Select
+          id='KnownHosts-select'
+          labelId='KnownHosts-label'
+          label='Host'
+          margin='dense'
+          name='host'
+          value={hostsState.selectedHost}
+          fullWidth={true}
+          onChange={e => selectHost(e.target.value)}
+        >
+          <Button value={hostsState.selectedHost} onClick={openAddKnownHostDialog}>
+            <span>Add new host</span>
+            <AddIcon fontSize='small' color='primary' />
+          </Button>
+
+          {
+            hostsState.hosts.map((host, index) => (
+              <MenuItem className='KnownHosts-item' value={host} key={index}>
+                <div className='KnownHosts-item__label'>
+                  <Check />
+                  <span>{host.name} ({ getHostPort(hostsState.hosts[index]).host }:{getHostPort(hostsState.hosts[index]).port})</span>
+                </div>
+
+                { host.editable && (
+                  <IconButton className='KnownHosts-item__edit' size='small' color='primary' onClick={(e) => {
+                    openEditKnownHostDialog(hostsState.hosts[index]);
+                  }}>
+                    <EditRoundedIcon fontSize='small' />
+                  </IconButton>
+                ) }
+              </MenuItem>
+            ))
+          }
+        </Select>
+      </FormControl>
+
+      <KnownHostDialog
+        isOpen={dialogState.open}
+        host={dialogState.edit}
```

Comment:
might've missed why this is called host = edit

Reply:
`edit` holds the host that's going to be edited (see openEditKnownHostDialog method)